### PR TITLE
Icon

### PR DESCRIPTION
**Pull request by david-gomey**
Create icon component
- FAIcon for FontAwesome icons
- Icon for /icons

### CODE REVIEW

File: src/Icon/types.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export interface IconProps extends React.SVGProps<SVGSVGElement> {
+  icon: (props: React.SVGProps<SVGSVGElement>) => JSX.Element;
+  size: Size;
+}
+
+export interface FAIconProps extends Omit<FontAwesomeIconProps, 'size'> {
+  color: ColorsTypes;
+  icon: IconDefinition;
+  size: Size;
```
Comment (EduardoRG) [inline src/Icon/types.ts:16]: I would consider setting size and color as optional props, with default values like:
```
size: "1x"
color: "inherit" // or just keep it undefined
```

What do you think?

**Review by EduardoRG — CHANGES_REQUESTED** [review]
I really like the way we can use this component, very good job!

File: src/Icon/icons/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export { analytics } from './analytics';
+export { calendly } from './calendly';
+export { dialogflow } from './dialogflow';
+export { formulas } from './formulas';
+export { googleSheets } from './googleSheets';
+export { hubspot } from './hubspot';
+export { nlu } from './nlu';
+export { zapier } from './zapier';
```
[the remark this answers is not in the document]
Comment (david-gomey) [reply src/Icon/icons/index.ts:8]: With your question I got the idea that we can try to change the configuration and import these icons in the following way:

import { calendly } from '@landbot/lui/icons';

@EduardoRG  If we can do this import, do you think it's important to change the name too?

[the remark this answers is not in the document]
Comment (EduardoRG) [reply src/Icon/icons/index.ts:8]: I like that! So I was wondering if it makes sense to export also from the `src/Icon/index.ts` file

File: src/Icon/types.ts (hunk 1 above)
Comment (david-gomey) [reply src/Icon/types.ts]: Changed with default values as: 

`
size: 1x;
color: blue.main
`

File: src/Icon/icons/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-gomey) [reply src/Icon/icons/index.ts:8]: Changes made to be able to do(with webpack 5):

`import { calendly } from '@landbot/lui/icons';`

@EduardoRG  Do you think we should also change the name of the icons?

**Review by amalv — CHANGES_REQUESTED** [review]

**Review by EduardoRG — APPROVED** [review]

**Review by amalv — APPROVED** [review]

File: src/Icon/FAIcon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+export default {
+  title: 'Components/Icon/FAIcon',
+  component: FAIcon,
+  argTypes: {
+    color: {
+      control: {
+        type: 'select',
+      },
+      defaultValue: 'blue.main',
```
Comment (amalv) [inline src/Icon/FAIcon.stories.tsx:18]: There is a warning regarding `defaultValue` being deprecated, in order to fix it the `defaultValue` would have to use the new API and set the default value like this:

```
 argTypes: {
    color: {
      control: {
        type: 'select',
      },
      ...
    },
},
args: {
  color: 'blue.main',
}
```
 
More info [here](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#no-longer-inferring-default-values-of-args)

Comment (david-gomey) [reply src/Icon/FAIcon.stories.tsx:18]: Nice, I will change it.

**Pull request closed without merging**